+++
draft = false
date = "2010-01-28T21:29:31-02:00"
title = "Another cause for buildout failures: system distributed Python"
keywords = "python, rants, buildout"
created = "2010-01-28T21:04:53-02:00"
modified = "2010-02-05T11:56:06-02:00"
permaLink = "http://play.pixelblaster.ro/blog/archive/2010/01/28/another-cause-for-buildout-failures-system-distributed-python"
author = "Tiberiu Ichim"
+++

<p>I've had a buildout bootstrap process failure, this time a weird one, perhaps I should document the bug and report it.</p>
<p>The latest Ubuntu version which I have installed (Lucid Lynx) comes with a package called python-pkg-resources, which packages <a class="external-link" href="http://www.python.org/dev/peps/pep-0365/">pkg_resources</a>, which used to be available only through the setuptools distribution. Buildout's bootstrap.py  tries to guess if Setuptools or Distribute are installed by checking the availability of pkg_resources; by guessing wrong it all comes to a crash at the end.</p>
<p>I'm not very interested in debugging these types of problems anymore. Distribution/packaging tools should just work. I want to focus on my work, not debug the toolchain. No more corner cases or whatver. So I'm gonna compile separate Pythons in the future, especially when dealing with older Zope/Plones.</p>
